include/net/net_pkt.h

```diff
@@ -83,6 +83,9 @@ struct net_pkt {
 	u8_t *appdata;	/* application data starts here */
 	u8_t *next_hdr;	/* where is the next header */
 
+	/** Reference counter */
+	atomic_t atomic_ref;
```

Comment:
Ok, dumb question, "The name of the element has been changed from "ref" to "atomic_ref" to
cause a compile error" - sounds good, but why it was moved around?

Reply:
Because there it is guaranteed to be aligned to 4 bytes (assuming pointers are 4 bytes).
There where it was, it would have resulted in padding depending on CONFIG_NET_STATISTICS and CONFIG_NET_TCP.

Reply:
> Because there it is guaranteed to be aligned to 4 bytes

Right, as it changes from u8_t to atomic word. Ok.